src/ppom-controller.ts

```diff
@@ -274,7 +274,20 @@ export class PPOMController extends BaseControllerV2<
       if (id === this.state.chainId) {
         return;
       }
-      let { chainStatus } = this.state;
+      let chainStatus = { ...this.state.chainStatus };
+      // delete ols chainId if total number of chainId is equal 5
+      const chainIds = Object.keys(chainStatus);
+      if (chainIds.length >= 5) {
+        const oldestChainId = chainIds.sort((c1, c2) =>
+          (chainStatus[c1]?.lastVisited as any) >
```

Comment:
NIT: since chainStatus[X].lastVisited will be a number, I think can we simply just do
```suggestion
          Number((chainStatus[c1]?.lastVisited) - Number((chainStatus[c2]?.lastVisited)
```

instead of the ternary operator.